Added very basic tutorial

aviklai opened the pull request.

danielhers commented on static/js/tutorial.js:
```diff
@@ -0,0 +1,28 @@
+              my: 'top center',
+              at: 'bottom center'
+            }, {
+              content: '<p>בפאנל הזה תוכלו לבחור את סוגי התאונות שאתם מעוניינים לראות ותווך התאריכים שלהן. בנוסף, תוכלו לצפות בפרטי התאונות שנמצאות בפריים הנוכחי.</p>',
```
תווך -> טווח
Better not use technical terms such as "panel" and "frame".

danielhers commented on templates/index.html:
```diff
@@ -10,17 +10,24 @@
             <link rel="stylesheet" href="{{ ASSET_URL }}" />
         {% endassets %}
-
+        <script src="../static/js/underscore-min.js"></script>
```
Aren't these included in the assets?

danielhers commented on templates/index.html:
```diff
@@ -385,6 +394,12 @@ <h5 class="type"></h5>
             ga('send', 'pageview');
         </script>
+	
+	<script type="text/javascript" src="../static/js/tutorial.js"> </script>
```
Better include in assets

aviklai replied:
I tried but got an webasset error. I moved all the css to the assets. The javascript files are already minified.

danielhers commented:
Great, thanks!
Please remove the following files:

    static/js/backbone-min.js
    static/js/tourist.min.js
    static/js/toutorial.js~
    static/js/tutorial.js~
    static/js/underscore-min.js

It would be better if you squash/amend your commits so that these files do not appear at all, rather than added and then removed.

Also, what are the changes to the following files?

    static/css/bootstrap.css
    static/css/bootstrap.css.map
    
Did you write the following files? If not, where did you get them from?

    static/js/tourist.js
    static/css/tourist.css

Better mention the issue number, #146, in the commit message. Also, comment on the issue if you start working on it.

aviklai commented:
I removed the linux backup files. 
About bootstrap - it is just the latest version of bootstrap  from:
http://getbootstrap.com/
I had to add the bootstrap.css.map file because got error from chrome otherwise.

about tourist.js and tourist.css:
http://easelinc.github.io/tourist/
A javascript library for web toutorials.

danielhers commented:
This is better, but the minified and map files are generated and should not be under version control. They should be minified by rjsmin if you include them in the assets. Let's try to solve that.

aviklai commented:
Ok, I compressed the tutorial dependencies js files with closure_js compressor. It was already imported to the project so I used it.

danielhers commented:
Sorry, not quite. closure.js was replaced by rjsmin because it doesn't require Java - pull from the upstream to see. Anyway, the minified files shouldn't be in your commit at all. The original files should be in the assets in index.html, and they will be minified dynamically.
And do you know how to squash[http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html] your commits? It's better than having many small commits reverting each other.

aviklai commented:
Like this? And about the squashing - I didn't know but from what I read at the link you provided, I can just do it for commits I still haven't pushed, right?

danielhers commented:
Looks good, but there are two commits, not one...
I just don't understand why you added the js assets in a separate block and not with the existing one (app.min.js). There's already backbone and underscore there...

aviklai commented:
Fixed, I squashed it to one commit. About the assets - I moved it to the assets you mentioned buy I change the filter to rjsmin because it was clousure.js

danielhers commented:
Great, now for some finer nuances (sorry, but we have high standards in the project): there's already backbone.js and underscore.js in static/js/libs. In general this is where we put libraries downloaded from the internet. No need to have them twice.
Also, the pull request can't be merged because there are conflicts. Please `pull --rebase upstream master` and resolve the conflicts, then push again to update the pull request.

danielhers commented:
Also, the issue number you should mention in the commit is #146.
#214 is this pull request.

aviklai commented:
O.K, I resolved the conflicts.

danielhers commented:
Excellent! I'm merging this :)
As further improvement, how about translating the buttons?

aviklai commented:
You mean to hebrew?

danielhers commented:
Of course :)

On Tue, Apr 28, 2015 at 9:53 AM aviklai <notifications@github.com> wrote:

> You mean to hebrew?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hasadna/anyway/pull/214#issuecomment-96949616>.
>

aviklai commented:
Done.
